independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/ResteasyReactiveRequestContext.java

```diff
@@ -796,6 +796,18 @@ public void initPathSegments() {
         }
     }
 
+    private static final String PRODUCES_CHECKED_PROPERTY_KEY = AbstractResteasyReactiveContext.CUSTOM_RR_PROPERTIES_PREFIX
+            + "ProducesChecked";
+
+    public void setProducesChecked(boolean checked) {
```

Comment:
This is introducing a `ConcurrentHashMap::put` there, but TBH it's still better than performing the check twice I think...
2 qq:
1. ClassRouting vs FixedProduces handler's are always running in that order? 
2. Do we have other mechanism to store this information as a plain field somewhere instead, in the context?

Reply:
> ClassRouting vs FixedProduces handler's are always running in that order?

Yes.

> Do we have other mechanism to store this information as a plain field somewhere instead, in the context?

We certainly can, but I was removing fields from the context because it was getting to big and we wanted to make it fit better in cache lines :)

Reply:
I can check with JOL, and if it is within some padding I am +1 to move it to a field, it sadly can matter in bench-marketing :"/  but need to be sure that's free